### DISCUSSION TITLE
Support datasources being excluded from Explore

### DISCUSSION THREAD

**Discussion by chrispappas** [grafana/grafana#58277] [Explore]
We are building a datasource to provide annotations only, where its data should not be queryable via Explore panel. To reduce the number of options presented to our users, we want to exclude this particular datasource from appearing in the Explore panel.

There are a number of options in the plugin config such as `annotations`, `metrics`, `alerting` etc. and I see in the Datasource Service implementation the [`getList` method](https://github.com/grafana/grafana/blob/main/public/app/features/plugins/datasource_srv.ts#L203) the application of filters based on this plugin metadata. That seems like an easy place to configure your datasource plugin to be excluded from the Explore tab.

My datasource also does not provide a QueryEditor component, perhaps the absence of that could be used as the signal to exclude it from the datasource list?

Open to further discussion on this item and potentially submitting some work to support as a PR.

**Comment by Elfo404** [comment]
that's an interesting idea. I guess if a plugin only has "annotations" support (and no metrics, logs, traces, etc...) we may be able to safely exclude it from the datasource list in Explore, although we should probably check if doing so won't accidentally remove also other datasources that have wrong info in the plugin.json.

**Reply by Elfo404** [reply]
On a side note, regarding 
> My datasource also does not provide a QueryEditor component, perhaps the absence of that could be used as the signal to exclude it from the datasource list?

a datasource can provide no editor but still be able to query data, so i wouldn't use that as a discriminator. I.E. imagine a custom built datasource that sends requests to an hardcoded api, or that maybe only needs datasource level configuration but has no customization options for queries.

**Comment by ifrost** [comment]
Does it apply only to Explore? Should such data source be excluded from Dashboards and Alerting?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
